syntax = "proto3";
package audio;

service Audio{
    rpc CreateAudio(CreateAudioRequest) returns (CreateAudioResponse){}
    rpc ReadAudio(ReadAudioRequest) returns (ReadAudioResponse){}
    rpc UpdateAudio(UpdateAudioRequest) returns (UpdateAudioResponse){}
    rpc DeleteAudio(DeleteAudioRequest) returns (DeleteAudioResponse){}
    rpc ReadAudioList(ReadAudioListRequest) returns (ReadAudioListResponse){}
}

// ===================== Utils ========================

message AudioObject{
    int32 id = 1;
    int32 channel_id = 2;
    string title = 3;
    string status = 4;
    string duration = 5;
    string url = 6;
    string updated_at = 7;
    string created_at = 8;
}

message NewAudioObject{
    string title = 1;
    int32 channel_id = 2;
    string status = 3;
    string duration = 4;
    string url = 5;
}

// =================== Request ===================

message CreateAudioRequest {
    NewAudioObject new_audio_object = 1;
}

message ReadAudioRequest {
    int32 id = 1;
}

message ReadAudioByTitleRequest {
    string title = 1;
}

message UpdateAudioRequest {
    int32 id = 1;
    NewAudioObject new_audio_object = 2;
}

message DeleteAudioRequest {
    int32 id = 1;
}

message ReadAudioListRequest {}

// ========================= Response ==========================


message CreateAudioResponse {
    string msg = 1;
}

message ReadAudioResponse {
    AudioObject audio_object = 1;
    string msg = 2;
}

message UpdateAudioResponse {
    string msg = 1;
}

message DeleteAudioResponse {
    string msg = 1;
}

message ReadAudioListResponse {
    repeated AudioObject audio_objects = 1;
    string msg = 2;
}


